pkg/client/fake/client.go

```diff
@@ -290,9 +293,22 @@ func (t versionedTracker) Add(obj runtime.Object) error {
 		if err != nil {
 			return fmt.Errorf("failed to get accessor for object: %w", err)
 		}
-		if accessor.GetDeletionTimestamp() != nil && len(accessor.GetFinalizers()) == 0 {
-			return fmt.Errorf("refusing to create obj %s with metadata.deletionTimestamp but no finalizers", accessor.GetName())
+
+		gvk, err := apiutil.GVKForObject(obj, t.scheme)
+		if err != nil {
+			panic(fmt.Errorf("failed to get gvk for object %T: %w", obj, err))
+		}
+
+		if !allowsGracefulDelete(gvk, obj) {
```

Comment:
That doesn't mean it will never delete the object but we can observe it at some certain time. The fake client can not simulate the graceful deletion now so I think it's reasonable to at least make it possible to observe such object for covering more cases within test cases.